### PR TITLE
Cli step 7 b06 correct

### PR DESCRIPTION
**Pull request by kmilo9999**
Adding CLI to step 7
- Files formatted and cleaned
- Added workflow for step7 CLI

### CODE REVIEW

File: src/icesat2_tracks/analysis_db/B06_correct_separate_var.py [hunk 1]
```diff
@@ -350,419 +143,592 @@ def define_noise_wavenumber_piecewise(data_xr, plot_flag=False):
...
+        )
+        gs = GridSpec(8, 3, wspace=0.1, hspace=1.5)
+
+        k_lims = G_gFT_wmean.k_lim
+        xlims = G_gFT_wmean.k[0], G_gFT_wmean.k[-1]
+        #
+        k = high_beams[0]
+        for pos, k, pflag in zip(
+            [gs[0:2, 0], gs[0:2, 1], gs[0:2, 2]], high_beams, [True, False, False]
+        ):
```
Comment (cpaniaguam) [inline src/icesat2_tracks/analysis_db/B06_correct_separate_var.py:342]: Some comments here would be nice to have!

File: src/icesat2_tracks/analysis_db/B06_correct_separate_var.py [hunk 2]
```diff
@@ -350,419 +143,592 @@ def define_noise_wavenumber_piecewise(data_xr, plot_flag=False):
...
+            color="black",
+        )
+        ax9.spines["left"].set_visible(False)
+        ax9.spines["right"].set_visible(True)
+        ax9.tick_params(labelright=True, right=True, labelleft=False, left=False)
+        ax9.axhline(10, linewidth=0.8, linestyle="--", color="black")
+        plt.xlabel("X (km)")
+
+        F.save_light(path=plot_path, name=str(track_name) + "_B06_atten_ov")
+        F.save_pup(path=plot_path, name=str(track_name) + "_B06_atten_ov")
```
Comment (cpaniaguam) [inline src/icesat2_tracks/analysis_db/B06_correct_separate_var.py:484]: Maybe another function for this plot?

File: src/icesat2_tracks/analysis_db/B06_correct_separate_var.py [hunk 3]
```diff
@@ -350,419 +143,592 @@ def define_noise_wavenumber_piecewise(data_xr, plot_flag=False):
...
+            height_data = np.interp(
+                dist_stencil, T3_sel["dist"], T3_sel["heights_c_weighted_mean"]
             )
-        else:
-            G_height_model_temp[str(i) + bb] = xr.DataArray(
-                Gx_1.y_model.data,
-                coords=Gx_1.coords,
-                dims=Gx_1.dims,
-                name="height_model",
+            return height_model, np.nan, dist_nanmask
```
Comment (cpaniaguam) [inline src/icesat2_tracks/analysis_db/B06_correct_separate_var.py:542]: Can these safely be put at the top?

File: src/icesat2_tracks/analysis_db/B06_correct_separate_var.py [hunk 4]
```diff
@@ -350,419 +143,592 @@ def define_noise_wavenumber_piecewise(data_xr, plot_flag=False):
...
+                    )
+                else:
+                    G_height_model_temp[str(i) + bb] = xr.DataArray(
+                        Gx_1.y_model.data,
+                        coords=Gx_1.coords,
+                        dims=Gx_1.dims,
+                        name="height_model",
+                    )
+
+            G_height_model[bb] = xr.concat(G_height_model_temp.values(), dim="x").T
```
Comment (cpaniaguam) [inline src/icesat2_tracks/analysis_db/B06_correct_separate_var.py:583]: This block seems like a good candidate to move to a function. What do you think?

Comment (cpaniaguam) [reply src/icesat2_tracks/analysis_db/B06_correct_separate_var.py:583]: Probably up to L587.

File: src/icesat2_tracks/analysis_db/B06_correct_separate_var.py [hunk 5]
```diff
@@ -350,419 +143,592 @@ def define_noise_wavenumber_piecewise(data_xr, plot_flag=False):
...
-    T3 = T3.sort_values("x")
-    T3 = T3.sort_values("dist")
-
-    T3["heights_c_model"] = np.interp(
-        T3["dist"], continous_x_grid, continous_height_model
-    )
-    T3["heights_c_model_err"] = np.interp(T3["dist"], continous_x_grid, concented_err)
-    T3["heights_c_residual"] = T3["heights_c_weighted_mean"] - T3["heights_c_model"]
+            B3_v2[bb] = T3
+            Gx_v2[bb] = Gx_k
```
Comment (cpaniaguam) [inline src/icesat2_tracks/analysis_db/B06_correct_separate_var.py:624]: This other block looks like it's its own entity.

File: src/icesat2_tracks/analysis_db/B06_correct_separate_var.py [hunk 6]
```diff
@@ -350,419 +143,592 @@ def define_noise_wavenumber_piecewise(data_xr, plot_flag=False):
...
 
-    B3_v2[bb] = T3
-    Gx_v2[bb] = Gx_k
+        try:
+            G_angle = xr.open_dataset(
+                load_path_angle + "/B05_" + track_name + "_angle_pdf.nc"
+            )
```
Comment (cpaniaguam) [inline src/icesat2_tracks/analysis_db/B06_correct_separate_var.py:629]: This is likely the error prone line. Consider using an [else block ](https://docs.python.org/3/tutorial/errors.html#:~:text=The%20try%20%E2%80%A6%20except%20statement%20has%20an%20optional%20else%20clause%2C%20which%2C%20when%20present%2C%20must%20follow%20all%20except%20clauses.%20It%20is%20useful%20for%20code%20that%20must%20be%20executed%20if%20the%20try%20clause%20does%20not%20raise%20an%20exception.%20For%20example%3A).

File: src/icesat2_tracks/analysis_db/B06_correct_separate_var.py [hunk 7]
```diff
@@ -350,419 +143,592 @@ def define_noise_wavenumber_piecewise(data_xr, plot_flag=False):
...
+            plt.pcolor(Ga_abs)
+            plt.xlabel("abs angle")
+            plt.ylabel("x")
+
+            ax = plt.subplot(2, 1, 2)
+            Ga_best.plot()
+            plt.title("angle front " + str(theta * 180 / np.pi), loc="left")
+            ax.axvline(theta, color="red")
+            F.save_light(path=plot_path, name="B06_angle_def")
+        except:
```
Comment (cpaniaguam) [inline src/icesat2_tracks/analysis_db/B06_correct_separate_var.py:666]: And here consider adding an exception. Maybe something like along the lines:
```python
try:
    # some code that might raise an exception
except ValueError as e:
    print(f"A ValueError occurred: {e}")
except Exception as e:
    print(f"An unexpected error occurred: {e}")
else:
    # code to run if no exception was raised
```

File: src/icesat2_tracks/analysis_db/B06_correct_separate_var.py [hunk 8]
```diff
@@ -350,419 +143,592 @@ def define_noise_wavenumber_piecewise(data_xr, plot_flag=False):
...
+        # %%
+        lam_p = 2 * np.pi / Gk.k
+        lam = lam_p * np.cos(theta)
+
+        if theta_flag:
+            k_corrected = 2 * np.pi / lam
+            x_corrected = Gk.x * np.cos(theta)
+        else:
+            k_corrected = 2 * np.pi / lam * np.nan
+            x_corrected = Gk.x * np.cos(theta) * np.nan
```
Comment (cpaniaguam) [inline src/icesat2_tracks/analysis_db/B06_correct_separate_var.py:680]: I would put this in a function also.

File: src/icesat2_tracks/analysis_db/B06_correct_separate_var.py [hunk 9]
```diff
@@ -350,419 +143,592 @@ def define_noise_wavenumber_piecewise(data_xr, plot_flag=False):
...
-
-Gk_v2 = Gk_v2.assign_coords(x_corrected=("x", x_corrected.data)).assign_coords(
-    k_corrected=("k", k_corrected.data)
-)
+        Gk_v2 = Gk_v2.assign_coords(x_corrected=("x", x_corrected.data)).assign_coords(
+            k_corrected=("k", k_corrected.data)
+        )
 
-Gk_v2.attrs["best_guess_incident_angle"] = theta
+        Gk_v2.attrs["best_guess_incident_angle"] = theta
```
Comment (cpaniaguam) [inline src/icesat2_tracks/analysis_db/B06_correct_separate_var.py:695]: This block seems to be for computing `Gk_v2`; I would put it in a function.

File: src/icesat2_tracks/analysis_db/B06_correct_separate_var.py [hunk 10]
```diff
@@ -350,419 +143,592 @@ def define_noise_wavenumber_piecewise(data_xr, plot_flag=False):
...
+        def save_table(data, tablename, save_path):
+            try:
+                io.save_pandas_table(data, tablename, save_path)
+            except Exception as e:
+                tabletoremove = save_path + tablename + ".h5"
+                echo(
+                    e, f"Failed to save table. Removing {tabletoremove} and re-trying.."
+                )
+                os.remove(tabletoremove)
+                io.save_pandas_table(data, tablename, save_path)
```
Comment (cpaniaguam) [inline src/icesat2_tracks/analysis_db/B06_correct_separate_var.py:712]: Can this function be move to the top?

**Review by cpaniaguam — CHANGES_REQUESTED** [review]
Great work on getting the cli set up and making the paths work! Thanks Camilo!!

I pointed out some opportunities for refactoring the code. Some reflect my personal style (like moving a block of code to a function) but a number of others will make the code more pythonic. Let me know what you think!

File: src/icesat2_tracks/analysis_db/B06_correct_separate_var.py [hunk 11]
```diff
@@ -298,7 +92,6 @@ def get_correct_breakpoint(pw_results):
 
 
 def get_breakingpoints(xx, dd):
-    import piecewise_regression
 
     x2, y2 = xx, dd
```
[the remark this answers is not in the document]
Comment (kmilo9999) [reply src/icesat2_tracks/analysis_db/B06_correct_separate_var.py:96]: I dont know if this is intentional or not, but it seems that `piecewise_regression.Fit` modifies the argument `x2`. The arguments the `get_breakingpoints` is called with are used later in the code and they need to be on a specific state.

File: src/icesat2_tracks/analysis_db/B06_correct_separate_var.py [hunk 12]
```diff
@@ -350,419 +143,592 @@ def define_noise_wavenumber_piecewise(data_xr, plot_flag=False):
...
+            Gk.where(~np.isnan(Gk["gFT_PSD_data"]), 0) * Gk["N_per_stancil"]
+        ).sum("beam") / Gk["N_per_stancil"].sum("beam")
+        G_gFT_wmean["N_photons"] = Gk["N_photons"].sum("beam")
 
-pos = gs[5:, 0:2]
-ax0 = F.fig.add_subplot(pos)
+        G_fft_wmean = (Gfft.where(~np.isnan(Gfft), 0) * Gfft["N_per_stancil"]).sum(
+            "beam"
+        ) / Gfft["N_per_stancil"].sum("beam")
+        G_fft_wmean["N_per_stancil"] = Gfft["N_per_stancil"].sum("beam")
```
[the remark this answers is not in the document]
Comment (kmilo9999) [reply src/icesat2_tracks/analysis_db/B06_correct_separate_var.py:223]: I just realized the `G_fft_wmean`  is not used in the rest of code. I am removing it, but still created the function.

File: src/icesat2_tracks/analysis_db/B06_correct_separate_var.py [hunk 13]
```diff
@@ -350,419 +143,592 @@ def define_noise_wavenumber_piecewise(data_xr, plot_flag=False):
...
+        gFT_PSD_data_error_mean = xr.concat(G_error_data.values(), dim="beam")
+
+        gFT_PSD_data_error_mean = (
+            gFT_PSD_data_error_mean.where(~np.isnan(gFT_PSD_data_error_mean), 0)
+            * Gk["N_per_stancil"]
+        ).sum("beam") / Gk["N_per_stancil"].sum("beam")
+        gFT_PSD_data_error_mean = (
+            gFT_PSD_data_error_mean.where(~np.isnan(gFT_PSD_data_error_mean), 0)
+            * Gk["N_per_stancil"]
+        ).sum("beam") / Gk["N_per_stancil"].sum("beam")
```
[the remark this answers is not in the document]
Comment (kmilo9999) [reply src/icesat2_tracks/analysis_db/B06_correct_separate_var.py:264]: This two assignments are the same. Therefore, the second operation is redundant ? Im removing it.

File: src/icesat2_tracks/analysis_db/B06_correct_separate_var.py [hunk 14]
```diff
@@ -350,419 +143,592 @@ def define_noise_wavenumber_piecewise(data_xr, plot_flag=False):
...
+                name="gFT_PSD_data_error",
+            ).expand_dims("beam")
+            G_error_data[bb] = xr.DataArray(
+                data=PSD_error_data,
+                coords=I.drop("N_per_stancil").coords,
+                name="gFT_PSD_data_error",
+            ).expand_dims("beam")
+
+        gFT_PSD_data_error_mean = xr.concat(G_error_model.values(), dim="beam")
+        gFT_PSD_data_error_mean = xr.concat(G_error_data.values(), dim="beam")
```
[the remark this answers is not in the document]
Comment (kmilo9999) [reply src/icesat2_tracks/analysis_db/B06_correct_separate_var.py:255]: I replaced it with a single line assignment

File: src/icesat2_tracks/analysis_db/B06_correct_separate_var.py [hunk 15]
```diff
@@ -350,419 +143,592 @@ def define_noise_wavenumber_piecewise(data_xr, plot_flag=False):
...
+            G_error_model[bb] = xr.DataArray(
+                data=PSD_error_model,
+                coords=I.drop("N_per_stancil").coords,
+                name="gFT_PSD_data_error",
+            ).expand_dims("beam")
+            G_error_data[bb] = xr.DataArray(
+                data=PSD_error_data,
+                coords=I.drop("N_per_stancil").coords,
+                name="gFT_PSD_data_error",
+            ).expand_dims("beam")
```
[the remark this answers is not in the document]
Comment (kmilo9999) [reply src/icesat2_tracks/analysis_db/B06_correct_separate_var.py:252]: I removed the second operation

File: src/icesat2_tracks/analysis_db/B06_correct_separate_var.py [hunk 16]
```diff
@@ -350,419 +143,592 @@ def define_noise_wavenumber_piecewise(data_xr, plot_flag=False):
...
+            k_end, pw_fit = define_noise_wavenumber_piecewise(
+                G_gFT_smth.sel(x=x) / k, plot_flag=False
+            )
 
-    decay = 0.5 - np.tanh((x - x_cutoff) / sigma_g) / 2
-    return decay
+            k_save = k_end_previous if k_end == k[0] else k_end
+            k_end_previous = k_save
+            k_lim_list.append(k_save)
+            echo("--------------------------")
```
[the remark this answers is not in the document]
Comment (kmilo9999) [reply src/icesat2_tracks/analysis_db/B06_correct_separate_var.py:306]: Love this code!!

File: src/icesat2_tracks/analysis_db/B06_correct_separate_var.py [hunk 17]
```diff
@@ -350,419 +143,592 @@ def define_noise_wavenumber_piecewise(data_xr, plot_flag=False):
...
+                .mean()
+            )
 
-    gFT_cos_coeff_sel = gFT_cos_coeff_sel * tanh_fitler(Gk_1.k, k_thresh, sigma_g=0.003)
-    gFT_sin_coeff_sel = gFT_sin_coeff_sel * tanh_fitler(Gk_1.k, k_thresh, sigma_g=0.003)
+            Gplot = Gplot.where(Gplot["N_per_stancil"] / Gplot["Lpoints"] >= 0.1)
+
+            alpha_range = iter(np.linspace(1, 0, Gplot.x.data.size))
+            for x in Gplot.x.data:
+                ialpha = next(alpha_range)
```
[the remark this answers is not in the document]
Comment (kmilo9999) [reply src/icesat2_tracks/analysis_db/B06_correct_separate_var.py:396]: You mean, taking the `ialpha = next(alpha_range)` assingment out of the loop?

File: src/icesat2_tracks/analysis_db/B06_correct_separate_var.py (hunk 3 above)
Comment (kmilo9999) [reply src/icesat2_tracks/analysis_db/B06_correct_separate_var.py:542]: Yes, they can. `fit_offset` is not used, I removed it.

File: src/icesat2_tracks/analysis_db/B06_correct_separate_var.py (hunk 15 above)
[the remark this answers is not in the document]
Comment (cpaniaguam) [reply src/icesat2_tracks/analysis_db/B06_correct_separate_var.py:252]: I think the operations the same but they seem to be defining different objects using different input. Namely, the first one is defining/updating `G_error_model[bb]`, while the second one does something to the same effect for `G_error_data[bb]`. Note also that the `data` argument is also different.

File: src/icesat2_tracks/analysis_db/B06_correct_separate_var.py (hunk 11 above)
[the remark this answers is not in the document]
Comment (cpaniaguam) [reply src/icesat2_tracks/analysis_db/B06_correct_separate_var.py:96]: Interesting.

File: src/icesat2_tracks/analysis_db/B06_correct_separate_var.py (hunk 17 above)
[the remark this answers is not in the document]
Comment (cpaniaguam) [reply src/icesat2_tracks/analysis_db/B06_correct_separate_var.py:396]: Correct!

[the remark this answers is not in the document]
Comment (cpaniaguam) [reply src/icesat2_tracks/analysis_db/B06_correct_separate_var.py:396]: Wait though: `next` is actually consuming the `alpha_range` itereator in every iteration so `ialpha` actually changes. It's better to use `zip` in the for loop:
```python
            for x, ialpha in zip(Gplot.x.data, alpha_range):
```

File: src/icesat2_tracks/analysis_db/B06_correct_separate_var.py (hunk 14 above)
[the remark this answers is not in the document]
Comment (cpaniaguam) [reply src/icesat2_tracks/analysis_db/B06_correct_separate_var.py:255]: Actually, it seems like `gFT_PSD_data_error_mean` is getting overwritten in L255. Are you deleting L254?

File: src/icesat2_tracks/app.py [hunk 18]
```diff
@@ -101,5 +112,15 @@ def make_b04_angle(  # TODO: revise naming @mochell
     run_job(_run_B04_angle, track_name, batch_key, ID_flag, output_dir, verbose)
 
 
+@app.command(help=_run_B04_angle.__doc__)
```
[the remark this answers is not in the document]
Comment (kmilo9999) [reply src/icesat2_tracks/app.py:115]: fixed

File: src/icesat2_tracks/analysis_db/B06_correct_separate_var.py [hunk 19]
```diff
@@ -25,230 +19,30 @@
...
-        plt.plot(np.log(data_xr.k), np.log(data_xr))
-        plt.plot(k_log, data_log)
-        plt.plot([np.log(k_end), np.log(k_end)], [-6, -5])
-    return k_end
 
-
-# new version
 def get_correct_breakpoint(pw_results):
     br_points = list()
     for i in pw_results.keys():
```
[the remark this answers is not in the document]
Comment (kmilo9999) [reply src/icesat2_tracks/analysis_db/B06_correct_separate_var.py:48]: I applied all these changes.

File: src/icesat2_tracks/analysis_db/B06_correct_separate_var.py (hunk 7 above)
Comment (kmilo9999) [reply src/icesat2_tracks/analysis_db/B06_correct_separate_var.py:666]: Done

File: src/icesat2_tracks/analysis_db/B06_correct_separate_var.py (hunk 9 above)
Comment (kmilo9999) [reply src/icesat2_tracks/analysis_db/B06_correct_separate_var.py:695]: #117

File: src/icesat2_tracks/analysis_db/B06_correct_separate_var.py (hunk 10 above)
Comment (kmilo9999) [reply src/icesat2_tracks/analysis_db/B06_correct_separate_var.py:712]: Done

File: src/icesat2_tracks/analysis_db/B06_correct_separate_var.py [hunk 20]
```diff
@@ -275,18 +69,18 @@ def get_correct_breakpoint(pw_results):
     alphas_v2_sorted = pd.Series(index=alphas_sorted, data=alphas_v2)
     br_points_sorted["breakpoint" + str(br_points_sorted.size + 1)] = "end"
```
[the remark this answers is not in the document]
Comment (kmilo9999) [reply src/icesat2_tracks/analysis_db/B06_correct_separate_var.py:70]: Done

File: src/icesat2_tracks/analysis_db/B06_correct_separate_var.py (hunk 11 above)
[the remark this answers is not in the document]
Comment (kmilo9999) [reply src/icesat2_tracks/analysis_db/B06_correct_separate_var.py:96]: Fixed

File: src/icesat2_tracks/analysis_db/B06_correct_separate_var.py [hunk 21]
```diff
@@ -350,419 +143,592 @@ def define_noise_wavenumber_piecewise(data_xr, plot_flag=False):
...
 
-k_lim_list = list()
-k_end_previous = np.nan
-x = G_gFT_smth.x.data[0]
-k = G_gFT_smth.k.data
+def run_B06_correct_separate_var(
+    track_name: str = typer.Option(..., callback=validate_track_name_steps_gt_1),
+    batch_key: str = typer.Option(..., callback=validate_batch_key),
+    ID_flag: bool = True,
+    output_dir: str = typer.Option(None, callback=validate_output_dir),
```
[the remark this answers is not in the document]
Comment (kmilo9999) [reply src/icesat2_tracks/analysis_db/B06_correct_separate_var.py:150]: Fixed

File: src/icesat2_tracks/analysis_db/B06_correct_separate_var.py [hunk 22]
```diff
@@ -350,419 +143,592 @@ def define_noise_wavenumber_piecewise(data_xr, plot_flag=False):
...
+        "ID_flag": ID_flag,
+        "output_dir": output_dir,
+    }
 
-font_for_pres()
-G_gFT_smth.coords["k_lim"] = ("x", k_lim_list)
-G_gFT_smth.k_lim.plot()
-k_lim_smth = G_gFT_smth.k_lim.rolling(x=3, center=True, min_periods=1).mean()
-k_lim_smth.plot(c="r")
+    report_input_parameters(**kargs)
```
[the remark this answers is not in the document]
Comment (kmilo9999) [reply src/icesat2_tracks/analysis_db/B06_correct_separate_var.py:173]: Fixed

File: src/icesat2_tracks/analysis_db/B06_correct_separate_var.py [hunk 23]
```diff
@@ -350,419 +143,592 @@ def define_noise_wavenumber_piecewise(data_xr, plot_flag=False):
...
-    Gplot = (
-        Gk.sel(beam=k)
-        .isel(x=slice(0, -1))
-        .gFT_PSD_data.squeeze()
-        .rolling(k=20, x=2, min_periods=1, center=True)
-        .mean()
-    )
+        B3 = dict()
+        for b in all_beams:
+            B3[b] = io.get_beam_hdf_store(B3_hdf5[b])
```
[the remark this answers is not in the document]
Comment (kmilo9999) [reply src/icesat2_tracks/analysis_db/B06_correct_separate_var.py:195]: Fixed

File: src/icesat2_tracks/analysis_db/B06_correct_separate_var.py [hunk 24]
```diff
@@ -350,419 +143,592 @@ def define_noise_wavenumber_piecewise(data_xr, plot_flag=False):
...
+            batch_key,
+            ID_flag,
+        ]  # init_from_input expects sys.argv with 4 elements
     )
 
-    k_save = k_end_previous if k_end == k[0] else k_end
-    k_end_previous = k_save
-    k_lim_list.append(k_save)
-    print("--------------------------")
+    kargs = {
```
[the remark this answers is not in the document]
Comment (kmilo9999) [reply src/icesat2_tracks/analysis_db/B06_correct_separate_var.py:166]: Fixed

File: src/icesat2_tracks/analysis_db/B06_correct_separate_var.py [hunk 25]
```diff
@@ -350,419 +143,592 @@ def define_noise_wavenumber_piecewise(data_xr, plot_flag=False):
...
-            Gplot.sel(x=x) / Gplot.k,
-            linewidth=0.5,
-            color=color_schemes.rels[k],
-            alpha=ialpha,
-        )
-        ax0.axvline(
-            k_lims.sel(x=x), linewidth=0.4, color="black", zorder=0, alpha=ialpha
-        )
+        h5_file_path = load_path_work / "B01_regrid" / (track_name + "_B01_binned.h5")
+        B3_hdf5 = h5py.File(h5_file_path, "r")
```
[the remark this answers is not in the document]
Comment (kmilo9999) [reply src/icesat2_tracks/analysis_db/B06_correct_separate_var.py:189]: Fixed

File: src/icesat2_tracks/analysis_db/B06_correct_separate_var.py [hunk 26]
```diff
@@ -350,419 +143,592 @@ def define_noise_wavenumber_piecewise(data_xr, plot_flag=False):
...
-            linewidth=0.5,
-            color=color_schemes.rels[k],
-            alpha=ialpha,
+        B3_hdf5.close()
+
+        Gk = xr.open_dataset(
+            load_path_work / "B02_spectra" / ("B02_" + track_name + "_gFT_k.nc")
+        )
+        Gx = xr.open_dataset(
+            load_path_work / "B02_spectra" / ("B02_" + track_name + "_gFT_x.nc")
```
[the remark this answers is not in the document]
Comment (kmilo9999) [reply src/icesat2_tracks/analysis_db/B06_correct_separate_var.py:203]: Fixed

File: src/icesat2_tracks/analysis_db/B06_correct_separate_var.py (hunk 12 above)
[the remark this answers is not in the document]
Comment (kmilo9999) [reply src/icesat2_tracks/analysis_db/B06_correct_separate_var.py:223]: Done

File: src/icesat2_tracks/analysis_db/B06_correct_separate_var.py (hunk 13 above)
[the remark this answers is not in the document]
Comment (kmilo9999) [reply src/icesat2_tracks/analysis_db/B06_correct_separate_var.py:264]: Done

File: src/icesat2_tracks/analysis_db/B06_correct_separate_var.py (hunk 14 above)
[the remark this answers is not in the document]
Comment (kmilo9999) [reply src/icesat2_tracks/analysis_db/B06_correct_separate_var.py:255]: Fixed. We removed the first assignment

File: src/icesat2_tracks/analysis_db/B06_correct_separate_var.py (hunk 15 above)
[the remark this answers is not in the document]
Comment (kmilo9999) [reply src/icesat2_tracks/analysis_db/B06_correct_separate_var.py:252]: removed the first assignment

File: src/icesat2_tracks/analysis_db/B06_correct_separate_var.py (hunk 16 above)
[the remark this answers is not in the document]
Comment (kmilo9999) [reply src/icesat2_tracks/analysis_db/B06_correct_separate_var.py:306]: Done

File: src/icesat2_tracks/analysis_db/B06_correct_separate_var.py (hunk 17 above)
[the remark this answers is not in the document]
Comment (kmilo9999) [reply src/icesat2_tracks/analysis_db/B06_correct_separate_var.py:396]: Done

File: src/icesat2_tracks/analysis_db/B06_correct_separate_var.py (hunk 1 above)
Comment (kmilo9999) [reply src/icesat2_tracks/analysis_db/B06_correct_separate_var.py:342]: It's unclear what this piece of code does.

File: src/icesat2_tracks/analysis_db/B06_correct_separate_var.py (hunk 20 above)
[the remark this answers is not in the document]
Comment (kmilo9999) [reply src/icesat2_tracks/analysis_db/B06_correct_separate_var.py:70]: Done

File: src/icesat2_tracks/analysis_db/B06_correct_separate_var.py [hunk 27]
```diff
@@ -275,18 +69,18 @@ def get_correct_breakpoint(pw_results):
...
     else:
         # take steepest slope
         alpah_v2_sub = alphas_v2_sorted[slope_mask]
-        print(alpah_v2_sub)
-        print(alpah_v2_sub.argmin())
+        echo(alpah_v2_sub)
+        echo(alpah_v2_sub.argmin())
         break_point_name = alpah_v2_sub.index[alpah_v2_sub.argmin()].replace(
             "alpha", "breakpoint"
         )
```
[the remark this answers is not in the document]
Comment (kmilo9999) [reply src/icesat2_tracks/analysis_db/B06_correct_separate_var.py:86]: Unclear what the code does

File: src/icesat2_tracks/analysis_db/B06_correct_separate_var.py [hunk 28]
```diff
@@ -333,11 +126,11 @@ def define_noise_wavenumber_piecewise(data_xr, plot_flag=False):
 
     pw_fit, breakpoint_log = get_breakingpoints(k_log, data_log.data)
```
[the remark this answers is not in the document]
Comment (kmilo9999) [reply src/icesat2_tracks/analysis_db/B06_correct_separate_var.py:127]: Done

File: src/icesat2_tracks/analysis_db/B06_correct_separate_var.py [hunk 29]
```diff
@@ -350,419 +143,592 @@ def define_noise_wavenumber_piecewise(data_xr, plot_flag=False):
...
+                Gk.sel(beam=k)
+                .isel(x=slice(0, -1))
+                .gFT_PSD_data.squeeze()
+                .rolling(k=20, x=2, min_periods=1, center=True)
+                .mean()
+            )
 
-    gFT_cos_coeff_sel = gFT_cos_coeff_sel * tanh_fitler(Gk_1.k, k_thresh, sigma_g=0.003)
-    gFT_sin_coeff_sel = gFT_sin_coeff_sel * tanh_fitler(Gk_1.k, k_thresh, sigma_g=0.003)
+            Gplot = Gplot.where(Gplot["N_per_stancil"] / Gplot["Lpoints"] >= 0.1)
```
[the remark this answers is not in the document]
Comment (kmilo9999) [reply src/icesat2_tracks/analysis_db/B06_correct_separate_var.py:392]: TODO #119

File: src/icesat2_tracks/analysis_db/B06_correct_separate_var.py (hunk 3 above)
Comment (kmilo9999) [reply src/icesat2_tracks/analysis_db/B06_correct_separate_var.py:542]: Done

File: .github/workflows/test-B01_SL_load_single_file.yml
[the remark this answers is not in the document]
Comment (kmilo9999) [reply .github/workflows/test-B01_SL_load_single_file.yml:1]: done

File: src/icesat2_tracks/analysis_db/B06_correct_separate_var.py (hunk 4 above)
Comment (kmilo9999) [reply src/icesat2_tracks/analysis_db/B06_correct_separate_var.py:583]: TODO

Comment (kmilo9999) [reply src/icesat2_tracks/analysis_db/B06_correct_separate_var.py:583]: #120

File: src/icesat2_tracks/analysis_db/B06_correct_separate_var.py (hunk 5 above)
Comment (kmilo9999) [reply src/icesat2_tracks/analysis_db/B06_correct_separate_var.py:624]: TODO

Comment (kmilo9999) [reply src/icesat2_tracks/analysis_db/B06_correct_separate_var.py:624]: #121

File: src/icesat2_tracks/analysis_db/B06_correct_separate_var.py (hunk 6 above)
Comment (kmilo9999) [reply src/icesat2_tracks/analysis_db/B06_correct_separate_var.py:629]: Done

Comment (kmilo9999) [reply src/icesat2_tracks/analysis_db/B06_correct_separate_var.py:629]: Done

File: src/icesat2_tracks/analysis_db/B06_correct_separate_var.py (hunk 8 above)
Comment (kmilo9999) [reply src/icesat2_tracks/analysis_db/B06_correct_separate_var.py:680]: Done

File: src/icesat2_tracks/analysis_db/B06_correct_separate_var.py (hunk 9 above)
Comment (kmilo9999) [reply src/icesat2_tracks/analysis_db/B06_correct_separate_var.py:695]: #122

**Review by cpaniaguam — APPROVED** [review]
Great work, thanks!

**Pull request merged**